backend/src/controllers/applicationController.ts

```diff
@@ -58,7 +145,25 @@ export const updateApplicationByID = asyncHandler(async (req, res, next) => {
 //  @throws {404} - If application not found
 //  @throws {400} - If ID is invalid
 export const deleteApplicationByID = asyncHandler(async (req, res, next) => {
-  const { id } = req.params;
+  const errors = validationResult(req);
+  if (!errors.isEmpty()) {
+    return next(createHttpError(400, validationErrorParser(errors)));
+  }
+
+  // Extract the validated 'id' from request parameters
+  const { id } = matchedData(req, { locations: ["params"] }) as { id: string };
+
+  // Find and delete the application by ID
+  const application = await Application.findByIdAndDelete(id).lean().exec();
+
+  if (!application) {
+    return next(createHttpError(404, "Application not found."));
+  }
+
+  res.status(200).json({
+    message: "Application deleted successfully.",
+    application,
```

Comment:
Probably isn't too big of a deal, but right now we don't any plans of sending the deleted application back to the frontend simply because it's not necessary and it might make the performance better by a tiny bit. Could you remove it for now? We can add it back if its needed in the future.

Reply:
Oh shoot will change that for sure